[RM Ch03] Storage resource description (#1852)
cntt-n/CNTT: doc/ref_model/chapters/chapter03.md

<a name="3.2.3"></a>
### 3.2.3 Storage

> A workload can request different type of data availibility : persistent or ephemeral storage.

Reviewer: availability is typically in terms of percentage

```suggestion
A workload can request different types of storage based on data longevity: persistent or ephemeral storage.
```

Author: @pgoyal01 Thanks, adopted!

Persistent storage outlives the compute instance whereas ephemeral storage is linked to compute instance lifecycle.

> Performance requirements are multiple depending on the workload to be hosted:low latency, IOPS oriented and capacity oriented storage. 

Reviewer: ```suggestion
There are multiple storage performance requirements such as latency, IOPS and capacity. For example, a workload may require one of its storage device to provide low latency, high IOPS and very large/hige storage capacity (terabytes of data).
```

Reviewer: May I suggest that capacity be separated from performance -- see storage comment below

Author: @pgoyal01 Thanks, suggestion adopted
I will modify the typo hige->huge

Low Latency storage is for workloads which have strong constraints on the time to access the storage.

> IOPS oriented storage is for workloads requiring lots of read/write actions.

Reviewer: ```suggestion
High IOPS oriented storage is for workloads requiring lots of read/write actions.
```

Author: @pgoyal01 Thanks, adopted!

Capacity oriented storage is for workloads that need lots of volumetry without strong perfomance constraints.

> Storage resources have the following attributes:

Reviewer: do we also need a device type (HDD, SSD, ...)?
do we need to include SDS (e.g., ceph)?

Author: @pgoyal01 I think the chapter 3 must keep only the modelling aspect without going into technologies. For me, it is not the good place to describe the device types and SDS. 
But you are right these topics must be addressed.  Can it be integrated into chapter 4, adding a section under 4.1.4 internal Infra capabilities?

Reviewer 2: I agree with @karinesevilla that such details would be better to have in Chapter 4.  @pgoyal01, if you agree, I would ask @karinesevilla to raise an issue to create a subsection 0f 4.1.4 to deal with the technology aspects of storage

Reviewer: @walterkozlowski @karinesevilla I agree with the approach to raise an issue and insert this in Ch4. Thanks

| Attribute           | Description                                                              |
|---------------------|--------------------------------------------------------------------------|
| `name`              | name of storage resources                                                |
| `data availibilty`  | persistent or ephemeral                                                  |  

> | `performance`       | low latency, IOPS oriented, capacity oriented                            |

Reviewer: ```suggestion
| `performance`       | latency, IOPS, capacity                            |
```

Author: @pgoyal01 Adopted

| `enhanced features` | replication, encryption                                                  |
| `type`              | block, object or file                                                    |

> | `size`              | size of disc in GB                                                       |

Reviewer: ```suggestion
| `size`              | size in GB                                                       |
```

Author: @pgoyal01 Adopted
